doc/how-to-freeze-versions.rst

```diff
@@ -0,0 +1,247 @@
+    their set of boot libraries as dependencies.
+
+    .. code-block:: cabal
+
+        executable unconstrained-base-exe
+        build-depends: base >0
+
+    .. code-block:: diff
+
+        $ cabal freeze --with-compiler=ghc-8.10.7
+        $ mv cabal.project.freeze ghc-8.10.7.freeze
+
+        $ cabal freeze --with-compiler=ghc-9.10.1
+        $ mv cabal.project.freeze ghc-9.10.1.freeze
+
+        $ cabal freeze --with-compiler=ghc-9.12.1
+        $ mv cabal.project.freeze ghc-9.12.1.freeze
+
+        $ diff ghc-8.10.7.freeze ghc-9.12.1.freeze --unified
+        -constraints: any.base ==4.14.3.0,
+        -             any.ghc-prim ==0.6.1,
+        -             any.integer-gmp ==1.0.3.0,
+        -             any.rts ==1.0.1
+        +constraints: any.base ==4.21.0.0,
+        +             any.ghc-bignum ==1.3,
+        +             any.ghc-internal ==9.1201.0,
+        +             any.ghc-prim ==0.13.0,
+        +             any.rts ==1.0.2
+
+        $ diff ghc-9.10.1.freeze ghc-9.12.1.freeze --unified
+        -constraints: any.base ==4.20.0.0,
+        +constraints: any.base ==4.21.0.0,
+                      any.ghc-bignum ==1.3,
+        -             any.ghc-internal ==9.1001.0,
+        -             any.ghc-prim ==0.11.0,
+        +             any.ghc-internal ==9.1201.0,
+        +             any.ghc-prim ==0.13.0,
+                      any.rts ==1.0.2
+
+Do you need to freeze?
+^^^^^^^^^^^^^^^^^^^^^^
+
+Why would you want to freeze? Don't we want to get minor updates of our
+dependencies, or at least patches, as soon as we can?  Well, although they
+shouldn't, it is possible that any kind of update introduces new bugs,
+performance issues, or some other kind of unexpected behaviour.  This is where
+``cabal.project.freeze`` comes in, as it ensures that dependencies don't
+unexpectedly change.  You can still update your dependencies, but you have to do
+it on purpose, by modifying or by deleting and regenerating
+``cabal.project.freeze`` file, and in the meantime you are guaranteed no
+surprises will happen.
+
+This consistency can be valuable as it ensures that all teammates, deployments,
+and continuous integration are installing the exactly same dependencies.  So if
+you are running and testing the code on your local machine, you are guaranteed
+that your teammate and your continuous integration will be running the exact
+same code, and that at the end that exact same code will get deployed.
+
+A ``.freeze`` file can be good to have when developing for yourself or within a
+private team.  If anyone using it can somehow have different inputs to
+the solver then the ``.freeze`` file can be troublesome.  It can prevent the
+solver from finding a different version of a dependency that would satisfy a
+different architecture or a different compiler version and boot libraries.
+
+.. Warning::
+
+    If publishing a package to Hackage, not matter what kind of component it
```

Comment:
```suggestion
    If publishing a package to Hackage, no matter what kind of component your project
```

Reply:
Thanks for spotting this typo @langston-barrett. I took the "no matter". The "it" here refers to the package and this, I kept.